cache/lru_cache.cc

```diff
@@ -287,10 +287,10 @@ void LRUCacheShard::Release(Cache::Handle* handle) {
     }
     if (e->refs == 1 && e->InCache()) {
       // The item is still in cache, and nobody else holds a reference to it
-      if (usage_ > capacity_) {
+      if (usage_ > capacity_ || force_erase) {
         // the cache is full
         // The LRU list must be empty since the cache is full
-        assert(lru_.next == &lru_);
+        assert(!(usage_ > capacity_) || lru_.next == &lru_);
```

Comment:
That would be less strict that the current assert because even when force_erase it might be the case that usage_ > capacity_ and we do not want to skip lru_.next == &lru_ for that.